pytorch_lightning/loggers/neptune.py

```diff
@@ -46,247 +93,469 @@ class NeptuneLogger(LightningLoggerBase):
-            project_name="shared/pytorch-lightning-integration",
-            experiment_name="default",  # Optional,
-            params={"max_epochs": 10},  # Optional,
-            tags=["pytorch-lightning", "mlp"],  # Optional,
-        )
-        trainer = Trainer(max_epochs=10, logger=neptune_logger)
+    **Quickstart**
 
-    **OFFLINE MODE**
+    Pass NeptuneLogger instance to the Trainer to log metadata with Neptune:
 
     .. testcode::
 
+        from pytorch_lightning import Trainer
         from pytorch_lightning.loggers import NeptuneLogger
 
-        # arguments made to NeptuneLogger are passed on to the neptune.experiments.Experiment class
         neptune_logger = NeptuneLogger(
-            offline_mode=True,
-            project_name="USER_NAME/PROJECT_NAME",
-            experiment_name="default",  # Optional,
-            params={"max_epochs": 10},  # Optional,
-            tags=["pytorch-lightning", "mlp"],  # Optional,
+            api_key="ANONYMOUS",                             # replace with your own
+            project="common/pytorch-lightning-integration",  # format "<WORKSPACE/PROJECT>"
+            tags=["training", "resnet"],                     # optional
         )
         trainer = Trainer(max_epochs=10, logger=neptune_logger)
 
-    Use the logger anywhere in you :class:`~pytorch_lightning.core.lightning.LightningModule` as follows:
+    **How to use NeptuneLogger?**
+
+    Use the logger anywhere in your :class:`~pytorch_lightning.core.lightning.LightningModule` as follows:
 
     .. code-block:: python
 
+        from neptune.new.types import File
+        from pytorch_lightning import LightningModule
+
         class LitModel(LightningModule):
             def training_step(self, batch, batch_idx):
                 # log metrics
-                self.logger.experiment.log_metric("acc_train", ...)
-                # log images
-                self.logger.experiment.log_image("worse_predictions", ...)
-                # log model checkpoint
-                self.logger.experiment.log_artifact("model_checkpoint.pt", ...)
-                self.logger.experiment.whatever_neptune_supports(...)
+                acc = ...
+                self.log("train/loss", loss)
 
             def any_lightning_module_function_or_hook(self):
-                self.logger.experiment.log_metric("acc_train", ...)
-                self.logger.experiment.log_image("worse_predictions", ...)
-                self.logger.experiment.log_artifact("model_checkpoint.pt", ...)
-                self.logger.experiment.whatever_neptune_supports(...)
+                # log images
+                img = ...
+                self.logger.experiment["train/misclassified_images"].log(File.as_image(img))
+
+                # generic recipe
+                metadata = ...
+                self.logger.experiment["your/metadata/structure"].log(metadata)
+
+    Check `Logging metadata docs <https://docs.neptune.ai/you-should-know/logging-metadata>`_
```

Comment:
not a big deal, just nitpick: I suggest to put the extensive user guide into the neptune.rst docs and keep all the docs here concise and focused around the api. but up to you

Reply:
@kamil-kaczmarek @Herudaio

Reply:
@awaelchli happy to move more extensive user guide to the neptune.rst. Would you mind to recommend a location of it in the PTL docs? Seems that `tutorials` is good candidate.

Reply:
we can do in a follow up if necessary, shouldn't block this pr

Reply:
@awaelchli agree, let not block this pr. I will update it after the release. Maybe you can include it in the weekly patch release.